kernelci.org/content/en/architecture/_index.md

```diff
@@ -9,19 +9,19 @@ description: "Learn the inner details behind the KernelCI systems"
 The first thing worth noting here is that there are two main parts of the
 overall KernelCI architecture:
 
-### [API & Pipeline](../api_pipeline)
+### [Maestro](../maestro)
 
-The top half of this picture shows native services interacting directly with
-the API: LAVA test labs, Kubernetes clusters, custom test environments and the
-job scheduler.  These are referred to as the KernelCI Pipeline in a loose
-sense.  Such services can be run pretty much anywhere, they are just API
-clients with a particular purpose.
+The top left box of this picture shows Maestro. While Maestro is only one software system, architecturally its flow has 2 parts: (1) triggering builds and list of tests to run on each platform and (2) driving such tests.
+
+Maestro has a [pipeline-based](../maestro/pipeline) design and an [API](../maestro/api/) that allow other systems and tools to interact with it to subscribe to events (such as new build/test triggers) or send patches to be tested by the KernelCI infrastructure.
+
+### CI Ecosystem
+
+Any CI/test system can be part of KernelCI by receiving triggers from Maestro and submitting their data to KCIDB. Receiving triggers is an optional for CI/test system, as many have their own trigger configuration already.
 
 ### [KCIDB](../kcidb)
 
-Other fully autonomous systems producing their own kernel builds and running
-their own tests can submit results to
-[KCIDB](../kcidb), which is a
+CI systems producing their own kernel builds and running their own tests can submit results to [KCIDB](../kcidb), which is a
 [BigQuery](https://cloud.google.com/bigquery) database to provide a unified
```

Comment:
I think it is not BigQuery anymore

Reply:
We have BigQuery, but it's not suitable for querying right now, and is going to be used for archiving and analytics (once we make the schema cheaper).

Reply:
I removed the mention to BigQuery.